### ISSUE TITLE
[feature] Provide stubs for Python API

### ISSUE THREAD

**Issue by marlamb** [conan-io/conan#17454]
### What is your suggestion?

Hi, I started using the [API](https://docs.conan.io/2/reference/extensions/python_api.html) provided by Conan2 and enjoy it a lot. However, I am missing stubs, which would bring multiple benefits

* Simpler coding using auto-completion offered by IDEs.
* Enables static checks, e.g. via `mypy` or `pyright`, such that users can also easily detect changes in new versions, even if not covered by tests.

I think at least the first point would also apply to all kind of functionality used in conanfiles, but for the API I would consider them even more beneficial, as one can safely assume that a growing number of users will have custom code relying on the API and is not that formally restricted than conanfiles are.

Note that already others tried to provide some [stubs for conanfiles](https://pypi.org/project/conan-stubs/), but according to their documentation it is for version 1.59 only. Even if there would be more activity, providing them from directly within the package would be great, as it would not suffer from any outdate-ness problem.

### Have you read the CONTRIBUTING guide?

- [x] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @marlamb 

Thanks for your suggestion.

I think this is still a bit too far away, there are many other steps that need to be done first:

- Stabilization and further public documentation of subapis
- Proper modeling and interface for model objects like Remote, Graph, etc
- Better (or some) type annotations in the interfaces

Note that most of the API is still undocumented, and this is because we didn't have enough time, it is just a matter of priorities. And while this would be a convenient thing, it is just a "nice to have" compared with many other aspects or features that are very necessary from the functional point of view, like workspaces.

So I am afraid that this will not be happening any time soon, sorry! (we will try to improve over the above points in the short and mid term future)

**Comment by marlamb** [comment]
Very much appreciated. I fully understand that it is not the highest priority as of now, nevertheless looking forward to it at some point in time :+1: .
